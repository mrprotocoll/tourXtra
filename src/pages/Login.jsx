import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { storeSession } from 'util/auth';

const Login = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();

    const res = await fetch(`${process.env.REACT_APP_API_URL}/login`, {
      method: 'POST',
      body: JSON.stringify({
        email,
        password,
      }),
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    });

    if (res.ok) {
      const data = await res.json();

      if (data.error) {
        setError(data.error);
      } else {
        storeSession(data.user, data.token);

        setEmail('');
        setPassword('');

        navigate('/');
      }
    } else {
      setError('Oops something went wrong. Try again.');
    }
  };

  return (
    <div className="auth_container">
      <form className="auth_form" onSubmit={handleSubmit}>

        <h1 className="text-xl font-black mb-5">Login</h1>

        {error && <div className="bg-red-300 p-4">{error}</div>}

        <input type="email" className="form_input" onChange={(element) => setEmail(element.target.value)} placeholder="Email" value={email} required />

        <input type="password" className="form_input" onChange={(element) => setPassword(element.target.value)} placeholder="Password" value={password} required />

        <div className="flex justify-between items-center">
          <button className="outline_btn" type="submit">Login</button>
          <div className="alt">
            Don&apos;t have an account?&nbsp;
            <a href="/signup" className="text-primary">
              Sign up
            </a>
          </div>
        </div>

      </form>
    </div>
  );
};

export default Login;
