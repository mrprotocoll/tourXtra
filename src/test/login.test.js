import React from 'react';

import renderer from 'react-test-renderer';

import { BrowserRouter } from 'react-router-dom';

import Login from 'pages/Login';

describe('Tests Login component', () => {
  it('Should render Navbar Component', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Login />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
