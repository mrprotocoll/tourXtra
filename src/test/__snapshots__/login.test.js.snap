// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Login component Should render Navbar Component 1`] = `
<div
  className="auth_container"
>
  <form
    className="auth_form"
    onSubmit={[Function]}
  >
    <h1
      className="text-xl font-black mb-5"
    >
      Login
    </h1>
    <input
      className="form_input"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      className="form_input"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <div
      className="flex justify-between items-center"
    >
      <button
        className="outline_btn"
        type="submit"
      >
        Login
      </button>
      <div
        className="alt"
      >
        Don't have an account?Â 
        <a
          className="text-primary"
          href="/signup"
        >
          Sign up
        </a>
      </div>
    </div>
  </form>
</div>
`;
